WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/CouponLineDetails.swift

```diff
@@ -67,15 +72,26 @@ struct CouponLineDetails: View {
                     }
                 }
                 ToolbarItem(placement: .primaryAction) {
-                    Button(Localization.done) {
-                        viewModel.saveData()
-                        presentation.wrappedValue.dismiss()
+                    if showValidateCouponLoading {
+                        ProgressView()
+                    } else {
+                        Button(Localization.done) {
```

Comment:
Maybe it doesn't work in a toolbar, but what about using the following?

```
Button(Localisation.done)
    .buttonStyle(SecondaryLoadingButtonStyle(isLoading: showValidateCouponLoading))
```

Reply:
Unfortunately, it adds an extra border:

<img src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/90c402d5-fb69-44bf-82e9-835e0c2e7ade" width="375">

But thanks for the suggestion, I didn't know about this component.